certora/specs/SocialRecoveryModule.spec

```diff
@@ -466,3 +466,29 @@ rule invalidatingNonceInRecovery(env e, address guardian, address[] newOwners, u
     bool isReverted = lastReverted;
     assert success => isReverted;
 }
+
+// This rule verifies that the safe can only add/revoke guardians and thresholds for itself, and not for other safe contracts.
+rule safeContractAccessToGuardiansAndThreshold(env e, address guardian, uint256 threshold, address otherSafeContract) {
+    requireGuardiansLinkedListIntegrity(guardian);
+    require e.msg.sender == safeContract;
+    require safeContract != otherSafeContract;
+
+    bool isGuardianInOtherSafeContract = currentContract.isGuardian(otherSafeContract, guardian);
+
+    currentContract.addGuardianWithThreshold(e, guardian, threshold);
```

Comment:
Can't we do a parametric call here instead of hardcoding `addGuardianWithThreshold`? Then it shows that this is true for *all* methods (and would prevent if a possible `addGuardiansWITHOUTThreshold()` method were added in the future, that it would miss a bug).

Reply:
Something similar to what was done here:

https://github.com/safe-global/safe-locking/blob/2d578f77fbfbd87318844252c3951cee26cbf2e0/certora/specs/SafeTokenLock.spec#L337-L351

Reply:
Good idea, have updated the rule 👍🏾